yuniql-platforms/oracle/OracleDataService.cs

```diff
@@ -0,0 +1,194 @@
+        {
+            this._connectionString = connectionString;
+        }
+
+        ///<inheritdoc/>
+        public bool IsTransactionalDdlSupported => false;
+
+        ///<inheritdoc/>
+        public bool IsSchemaSupported { get; } = true;
+
+        ///<inheritdoc/>
+        public bool IsBatchSqlSupported { get; } = true;
+
+        ///<inheritdoc/>
+        public bool IsUpsertSupported => false;
+
+        ///<inheritdoc/>
+        public string TableName { get; set; } = "__yuniql_schema_version";
+
+        ///<inheritdoc/>
+        public string SchemaName { get; set; }
+
+        ///<inheritdoc/>
+        public IDbConnection CreateConnection()
+        {
+            return new OracleConnection(_connectionString);
+        }
+
+        ///<inheritdoc/>
+        public IDbConnection CreateMasterConnection()
+        {
+            var masterConnectionStringBuilder = new OracleConnectionStringBuilder(_connectionString);
+            masterConnectionStringBuilder.DataSource = "INFORMATION_SCHEMA";
+
+            return new OracleConnection(masterConnectionStringBuilder.ConnectionString);
+        }
+
+        ///<inheritdoc/>
+        public List<string> BreakStatements(string sqlStatementRaw)
+        {
+            var sqlBatchParser = new SqlBatchParser(_traceService, new GoSqlBatchLineAnalyzer(), new CommentAnalyzer());
+            return sqlBatchParser.Parse(sqlStatementRaw).Select(s => s.BatchText).ToList();
+        }
+
+        ///<inheritdoc/>
+        public ConnectionInfo GetConnectionInfo()
+        {
+            var connectionStringBuilder = new OracleConnectionStringBuilder(_connectionString);
+            return new ConnectionInfo { DataSource = connectionStringBuilder.DataSource, Database = connectionStringBuilder.DataSource };
+        }
+
+        ///<inheritdoc/>
+        public string GetSqlForCheckIfDatabaseExists()
+            => @"
+SELECT 1 FROM SYS.ALL_TABLES WHERE OWNER = '${YUNIQL_DB_NAME}';
+            ";
+
+        ///<inheritdoc/>
+        public string GetSqlForCreateDatabase()
+            => throw new NotSupportedException("Create database is not supported in Oracle.");
+
+        ///<inheritdoc/>
+        public string GetSqlForCreateSchema()
+            => throw new NotSupportedException("Custom schema is not supported in Oracle.");
+
+        ///<inheritdoc/>
+        public string GetSqlForCheckIfDatabaseConfigured()
+            => @"
+SELECT 1 FROM SYS.ALL_TABLES WHERE OWNER = '${YUNIQL_DB_NAME}' AND TABLE_NAME = '${YUNIQL_TABLE_NAME}' AND ROWNUM = 1;
+            ";
+
+        ///<inheritdoc/>
+        public string GetSqlForCheckIfDatabaseConfiguredv10()
+            => @"
+SELECT 1 FROM SYS.ALL_TABLES WHERE OWNER = '${YUNIQL_DB_NAME}' AND TABLE_NAME = '__yuniqldbversion' AND ROWNUM = 1;
+            ";
+
+        ///<inheritdoc/>
+        public string GetSqlForConfigureDatabase()
+            => @"
+CREATE TABLE ${YUNIQL_TABLE_NAME} (
+	sequence_id NUMBER NOT NULL,
```

Comment:
We expect this column `sequence_id ` to be an AUTOINCREMENT column with SEED VALUE = 1. How we do this in Oracle?

Reply:
Hi, there are Sequencers in Oracle. We can create them as part of initial Schema create. https://www.techonthenet.com/oracle/sequences.php

Reply:
What about something like `id NUMBER GENERATED ALWAYS AS IDENTITY`. I just did quick google. Sequencers sounds little too much for this. I even thought of dropping this column as its has little use right now but kept anyway for backward compatibility.

Reply:
There is just one issue i guess, database compatibility. ```Oracle 12c introduced a new way that allows you to define an identity column for a table, which is similar to the AUTO_INCREMENT column in MySQL or IDENTITY column in SQL Server.```

Reply:
My suggestion is to only support versions Oracle 12c+.

Reply:
That's a problem then as we do need to support older versions and its minimal hit. Frankly, i like Oracle approach to be explicit about field value rather than just relying on the auto generated value. In this case, it looks better to have it auto-generated and as you said earlier this field currently have no use. I have seen just one ```UPDATE``` and thats based upon ``` version```.